File: php-advisor-core/AdvisorCore.php
<?php

namespace StormGeo\AdvisorCore;

/**
 * @package StormGeo\AdvisorCore
 */
class AdvisorCore
{
  /**
   * @var Chart
   */
  public $chart;

  /**
   * @var Climatology
   */
  public $climatology;

  /**
   * @var CurrentWeather
   */
  public $currentWeather;

  /**
   * @var Forecast
   */
  public $forecast;

  /**
   * @var Monitoring
   */
  public $monitoring;

  /**
   * @var Observed
   */
  public $observed;

  /**
   * @var Plan
   */
  public $plan;

  /**
   * @var Schema
   */
  public $schema;

  /**
   * @var Tms
   */
  public $tms;

  /**
   * @param   string  $token
   * @param   int     $retries
   * @param   int     $delay
   */
  public function __construct($token, $retries = 5, $delay = 5)
  {
    $this->chart = new Chart($token, $retries, $delay);
    $this->climatology = new Climatology($token, $retries, $delay);
    $this->currentWeather = new CurrentWeather($token, $retries, $delay);
    $this->forecast = new Forecast($token, $retries, $delay);
    $this->monitoring = new Monitoring($token, $retries, $delay);
    $this->observed = new Observed($token, $retries, $delay);
    $this->plan = new Plan($token, $retries, $delay);
    $this->schema = new Schema($token, $retries, $delay);
    $this->tms = new Tms($token, $retries, $delay);
  }
}

File: python-advisor-core/advisor_core.py
from request_handler import RequestHandler
from grouped_routes import (
    ForecastAPI,
    ObservedAPI,
    ClimatologyAPI,
    CurrentWeatherAPI,
    MonitoringAlertsAPI,
    PlanAPI,
    ChartAPI,
    TmsAPI,
    SchemaAPI
)

class AdvisorCore:
    def __init__(self, token, retries=5, delay=5):
        base_url="https://advisor-core.climatempo.io/api/"
        request_handler = RequestHandler(base_url, token, retries, delay)
        self.forecast = ForecastAPI(request_handler)
        self.observed = ObservedAPI(request_handler)
        self.current_weather = CurrentWeatherAPI(request_handler)
        self.climatology = ClimatologyAPI(request_handler)
        self.monitoring = MonitoringAlertsAPI(request_handler)
        self.plan = PlanAPI(request_handler)
        self.chart = ChartAPI(request_handler)
        self.tms = TmsAPI(request_handler)
        self.schema = SchemaAPI(request_handler)
